packages/core/server/admin-api/services/query-layer-decorator.ts

```diff
@@ -120,102 +64,51 @@ const decorator = (service: IDecoratedService) => ({
-    const entity = await service.update.call(this, uid, entityId, {
-      ...opts,
+    const entity = await service.findOne.call(this, uid, entityId, {
       populate: {
         ...relations.reduce((obj, key) => ({ ...obj, [key]: {} }), {}),
         url_alias: {
-          fields: ['id', 'generated'],
-        },
-        localizations: {
-          populate: {
-            url_alias: {
-              fields: ['id'],
-            },
-          },
+          fields: ['id', 'generated', 'url_path'],
         },
       },
     });
-
-    // Fetch the URL alias localizations.
-    const urlAliasLocalizations = entity.localizations
-      ?.map((loc) => loc.url_alias?.id)
-      ?.filter((loc) => loc) || [];
-
-    const entityWithoutLocalizations = {
-      ...entity,
-      localizations: undefined,
-    };
-
-    // If a URL alias is allready present, fetch it.
+  
+    // If a URL alias is already present, fetch it.
     if (opts.data.url_alias) {
       urlAliasEntity = await getPluginService('urlAliasService').findOne(opts.data.url_alias);
-      // @ts-ignore
     } else if (entity.url_alias) {
-      // @ts-ignore
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
       urlAliasEntity = entity.url_alias;
     }
-
+  
     // If a URL alias is present and 'generated' is set to false, do nothing.
     if (urlAliasEntity?.generated === false) {
       return service.update.call(this, uid, entityId, opts);
     }
-
+  
     // Generate the path.
-    const urlPattern = await getPluginService('urlPatternService').findByUid(uid, entity.locale);
-    const generatedPath = getPluginService('urlPatternService').resolvePattern(uid, entityWithoutLocalizations, urlPattern);
-
+    const generatedPath = getPluginService('urlPatternService').resolvePattern(uid, { ...entity, ...opts.data }, urlPattern);
+  
     // If a URL alias is present and 'generated' is set to true, update the alias.
     if (urlAliasEntity?.generated === true) {
-      urlAliasEntity = await getPluginService('urlAliasService').update(urlAliasEntity.id, {
-        url_path: generatedPath,
-        generated: true,
-        contenttype: uid,
-        // @ts-ignore
-        locale: entity.locale,
-        // @ts-ignore
-        localizations: urlAliasLocalizations,
-      });
+      urlAliasEntity = await getPluginService('urlAliasService').update(urlAliasEntity.id, { url_path: generatedPath, generated: true, contenttype: uid });
     }
-
+  
     // If no URL alias is present, create one.
     if (!urlAliasEntity) {
-      urlAliasEntity = await getPluginService('urlAliasService').create({
-        url_path: generatedPath,
-        generated: true,
-        contenttype: uid,
-        // @ts-ignore
-        locale: entity.locale,
-        // @ts-ignore
-        localizations: urlAliasLocalizations,
-      });
+      urlAliasEntity = await getPluginService('urlAliasService').create({ url_path: generatedPath, generated: true, contenttype: uid });
```

Comment:
All these changes to the the other decorators should not be nessecary.
Could you try to revert these changes and see it the tests are still successfull in the pipeline?